Cargo.toml

```diff
@@ -26,7 +26,7 @@ features = ["batch", "pkcs8"]
 
 [dependencies]
 curve25519-dalek = { version = "=4.0.0-rc.1", default-features = false, features = ["digest"] }
-ed25519 = { version = "2.1", default-features = false }
+ed25519 = { version = ">=2.2, <2.3", default-features = false }
```

Comment:
This kind of version range has been pretty detrimental to downstream users in the past. Was v2.2 _technically_ breaking?

Reply:
Look at the PR. There are several breaking changes to PKCS#8 support.

It's following the stated policy here:

https://github.com/RustCrypto/signatures/tree/master/ed25519#semver-policy

> The `pkcs8` module is exempted as it uses a pre-1.0 dependency, however, breaking changes to this module will be accompanied by a minor version bump.

As well as the one here:

https://github.com/dalek-cryptography/ed25519-dalek#public-api-semver-exemptions

Reply:
Ok got it. Why not specify the more future-friendly "2.2" dev here then?

Reply:
Future minor releases may make further breaking changes to the `pkcs8` crate.

Once the releases are out and it's known they don't make breaking changes, the requirement can be relaxed.

Reply:
Anyway, I wanted to understand the reasoning of explicit version declaration. Sorry if my approach felt confronting or unpleasent. If this type of declaration is meant to be clear and verbose so that other people reading would instantly understand (without the context of how cargo parses versioning) then that LGTM. Otherwise just curious on why such preference to dependency version constraints instead of the given shorthand.

Reply:
That's not quite how versioning works in this case. See [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2f20d4fb80bb6d1e03e726af492e03dc).

Reply:
Yeah, both `2.2` and `2.2.0` are interpreted as `>= 2.2.0, <3` under normal SemVer rules.

The use of versioning in place here is not specifically SemVer compatible, or that is to say, if you view the two links in my comment at the top of the thread the use of `pkcs8` is explicitly called out as being outside the scope of normal SemVer guarantees.

It's a compromise which lets part of the API get >1.x and get SemVer-compatible guarantees while still letting off-by-default features (which depend on pre-1.x crates) evolve at a different pace without forcing a major version bump every time.